webui/react/src/components/WorkspaceCreateModal.tsx

```diff
@@ -240,10 +316,24 @@ const WorkspaceCreateModalComponent: React.FC<Props> = ({ onClose, workspaceId }
           agentUserGroup?: V1AgentUserGroup;
           checkpointStorageConfig?: unknown;
           name: string;
+          clusterNamespacePairs?: { [key: string]: string };
         } = {
           name: workspaceName,
         };
 
+        const clusterNamespacePairs = Loadable.getOrElse([], resourceManagers).reduce(
+          (memo: Record<string, string>, name) => {
+            // check for undefined is important here because we don't want to add the element to the
+            // CreateWorkspace request if it is not defined. Without the check we add it as {key: ''}.
+            if (name in values && values[name] !== undefined) {
+              memo[name] = values[name]?.toString() ?? '';
+            }
+            return memo;
+          },
+          {},
+        );
+        body['clusterNamespacePairs'] = clusterNamespacePairs;
```

Comment:
here's the kicker with nesting the bindings -- because we now know our bindings can't be undefined and they're always strings, we can clean this code up:
```suggestion
        if (values.bindings) {
            body['clusterNamespacePairs'] = pick(values.bindings, resourceManagers.getOrElse([]));
        }
```

`pick` is a helper we get from lodash which is essentially
```tsx
pick = (obj: Record<string, unknown>, keys: string[]) => {
  return keys.reduce((memo: Record<string, unknown>, key) => {
    if (key in object) {
      memo[key] = object[key];
    }
    return memo;
  }, {});
}
```

Reply:
done!